go/store/nbs/store.go

```diff
@@ -978,68 +973,55 @@ func (nbs *NomsBlockStore) commit(ctx context.Context, current, last hash.Hash,
 	t1 := time.Now()
 	defer nbs.stats.CommitLatency.SampleTimeSince(t1)
 
-	anyPossiblyNovelChunks := func() bool {
-		nbs.mu.Lock()
-		defer nbs.mu.Unlock()
-		return nbs.mt != nil || len(nbs.tables.novel) > 0
-	}
+	nbs.mu.Lock()
+	nbs.waitForGC()
 
-	if !anyPossiblyNovelChunks() && current == last {
-		err := nbs.Rebase(ctx)
+	anyPossiblyNovelChunks := nbs.mt != nil || len(nbs.tables.novel) > 0
 
+	if !anyPossiblyNovelChunks && current == last {
+		nbs.mu.Unlock()
+		err := nbs.Rebase(ctx)
 		if err != nil {
 			return false, err
 		}
-
 		return true, nil
 	}
 
+	defer nbs.mu.Unlock()
+
 	// check for dangling references in |nbs.mt|
-	nbs.mu.Lock()
-	if err = nbs.errorIfDangling(checker); err != nil {
+	if err = nbs.errorIfDangling(current, checker); err != nil {
+		if errors.Is(err, ErrDanglingRef) {
+			nbs.mt = nil
```

Comment:
should we clear the memtable in `errorIfDangling`? is there ever a path where we don't want to clear it?